goal_src/engine/ui/progress/progress-part.gc

```diff
@@ -174,75 +151,62 @@
-            )
-        (+! a1-1 1)
-        )
+   (let ((a1-1 0))
+    (dotimes (a2-1 (-> *progress-process* 0 nb-of-particles))
+     (if (= (-> a0-1 matrix) (-> s5-0 particles a1-1 part matrix))
+      (set! s4-0 a1-1)
+      )
+     (+! a1-1 1)
+     )
+    )
+   (cond
+    ((= v1-0 3)
+     (if (= (-> s5-0 particle-state s4-0) 1)
+      (set! (-> arg2 vector 2 w) 42.0)
+      (set! (-> arg2 vector 2 w) 128.0)
       )
-    (cond
-      ((= v1-0 3)
-       (if (= (-> s5-0 particle-state s4-0) 1)
-           (set! (-> arg2 vector 2 w) 42.0)
-           (set! (-> arg2 vector 2 w) 128.0)
+     0
+     )
+    (else
+     (let ((s3-0 (new 'stack-no-clear 'vector))
+           (s2-0 (new 'stack-no-clear 'vector))
            )
-       0
-       )
-      (else
-        (let ((s3-0 (new 'stack-no-clear 'vector))
-              (s2-0 (new 'stack-no-clear 'vector))
-              )
-          (cond
-            ((or
-               (= s4-0 -1)
-               (= (-> s5-0 particle-state s4-0) 2)
-               (= (-> s5-0 particle-state s4-0) 1)
-               )
-             (set! (-> arg2 vector 0 x) 245760.0)
-             )
-            ((= v1-0 -1)
-             (set! (-> arg2 vector 0 x) 0.0)
-             (set! (-> arg2 vector 0 y) 0.0)
-             )
-            (else
-              ;; NOTE: this check has been added so we can run progress even when manipy isn't working yet.
-              (cond
-                ((nonzero? (-> s5-0 icons (logand s4-0 3) icon 0 node-list))
-                 (vector<-cspace!
-                   s3-0
-                   (-> s5-0 icons (logand s4-0 3) icon 0 node-list data v1-0)
-                   )
-                 (vector<-cspace!
-                   s2-0
-                   (-> s5-0 icons (logand s4-0 3) icon 0 node-list data 3)
-                   )
-                 (vector-! s3-0 s3-0 s2-0)
-                 (set! (-> arg2 vector 0 x) (* 48.0 (-> s3-0 x)))
-                 (set! (-> arg2 vector 0 y) (* 60.0 (-> s3-0 y)))
-                 (set! (-> arg2 vector 0 z) (* 32.0 (-> s3-0 z)))
-                 )
-                (else
-                  ;; NOTE: this is added as a temporary workaround.
-                  ;; if we don't have this stuff loaded right, just set the offset to 0.
-                  ;; this ends up being the right offset in all the cases except for things that
-                  ;; orbit the powercell. Those are just draw underneath the powercell 
-                  (set! (-> arg2 vector 0 x) 0.0)
-                  (set! (-> arg2 vector 0 y) 0.0)
-                  )
-                )
-              )
-            )
-          )
-        0
+      (cond
+       ((or
+         (= s4-0 -1)
+         (= (-> s5-0 particle-state s4-0) 2)
+         (= (-> s5-0 particle-state s4-0) 1)
+         )
+        (set! (-> arg2 vector 0 x) 245760.0)
+        )
+       ((= v1-0 -1)
+        (set! (-> arg2 vector 0 x) 0.0)
+        (set! (-> arg2 vector 0 y) 0.0)
         )
+       (else
```

Comment:
This `fuel-cell-progress-hud-orbit-callback` function has some changes so that it doesn't crash due to missing `manipy` stuff.  Can you replace this with the old version?